sycl/include/CL/sycl/device.hpp

```diff
@@ -214,14 +213,14 @@ class __SYCL_EXPORT device {
   friend T detail::createSyclObjFromImpl(decltype(T::impl) ImplObj);
 };
 
-} // namespace sycl
-} // __SYCL_INLINE_NAMESPACE(cl)
+} // __SYCL_OPEN_NS()
+__SYCL_CLOSE_NS()
 
 namespace std {
-template <> struct hash<cl::sycl::device> {
-  size_t operator()(const cl::sycl::device &Device) const {
-    return hash<std::shared_ptr<cl::sycl::detail::device_impl>>()(
-        cl::sycl::detail::getSyclObjImpl(Device));
+template <> struct hash<__sycl_ns_alias::device> {
```

Comment:
```suggestion
template <> struct hash<__sycl_ns_alias::device> {
```
Can we drop `_alias` from the `__sycl_ns_alias` and use `__sycl_ns` instead?